### ISSUE TITLE
Type error thrown on python 3.7 when trying to discover noticeboard

### ISSUE THREAD

**Issue by bryan-hoang** [DuncanMays/axon-ECRG#1]
While trying to use the library, the following line

```
axon_local_ips = await discovery.broadcast_discovery(num_hosts=1, port=config.comms_config.notice_board_port)
```
used to discover the notice board on the network automatically throws the following error:

```
$ src/worker.py
Exception ignored in: <function ClientSession.__del__ at 0xb58ee468>
Traceback (most recent call last):
  File "/home/bryan/.local/share/virtualenvs/mthe-493-group-a2-e9HO8nW2/lib/python3.7/site-packages/aiohttp/client.py", line 269, in __del__
    if not self.closed:
  File "/home/bryan/.local/share/virtualenvs/mthe-493-group-a2-e9HO8nW2/lib/python3.7/site-packages/aiohttp/client.py", line 894, in closed
    return self._connector is None or self._connector.closed
AttributeError: _connector
Traceback (most recent call last):
  File "src/worker.py", line 155, in <module>
    asyncio.run(main())
  File "/home/bryan/.asdf/installs/python/3.7.12/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/home/bryan/.asdf/installs/python/3.7.12/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "src/worker.py", line 142, in main
    axon_local_ips = await discovery.broadcast_discovery(num_hosts=1, port=config.comms_config.notice_board_port)
  File "/home/bryan/.local/share/virtualenvs/mthe-493-group-a2-e9HO8nW2/lib/python3.7/site-packages/axon/discovery.py", line 25, in broadcast_discovery
    async with aiohttp.ClientSession(conn_timeout=timeout) as session:
TypeError: __init__() got an unexpected keyword argument 'conn_timeout'
```

while running python 3.7. The error doesn't seem to occur on python 3.9.

**Comment by DuncanMays** [comment]
Hi Bryan, this is probably because the ClientSession class from the version of aiohttp for python 3.7 doesn't use the same keywords as the one in 3.9. I'm looking into this, and will push a fix as soon as one comes.

**Comment by bryan-hoang** [comment]
Nope. The only think that looks like it *might* be related is `@worker.rpc(comms_pattern='duplex', executor='Thread')`. Otherwise, we're not using those mentioned functions.

**Comment by DuncanMays** [comment]
It's definitely the keyword, I'm on 3.7 and getting: `DeprecationWarning: conn_timeout is deprecated, use timeout argument instead`. This is a problem because I can't take that argument out, and switching the keyword depending on the version of python or aiohttp is not something that I'm willing to do. Why are you guys using 3.7 again?

**Comment by bryan-hoang** [comment]
@jaredmcgrath made [this comment](https://github.com/bryan-hoang/mthe-493-group-a2/pull/7#issuecomment-1034391262) and brought up the option of using Python 3.7 so that we could use the prebuilt PyTorch wheels in https://github.com/ljk53/pytorch-rpi/, instead of having to use the wheel I compiled on my Raspberry Pi for python 3.9. @edan-parker also agreed to go in this direction, but we didn't know about axon's incompatibility with Python 3.7 before deciding.

Regarding [the article](https://medium.com/secure-and-private-ai-writing-challenge/a-step-by-step-guide-to-installing-pytorch-in-raspberry-pi-a1491bb80531) you sent me, those wheels only work for python 3.6.[^1], which would also lead to this issue.

> switching the keyword depending on the version of python or aiohttp is not something that I'm willing to do

Yep, I'd do the same. In that case, we may have to go with either sticking with a committed version of the wheel I compiled, or trying to host the wheel somewhere like in a GitHub release.

I wish there was a prebuilt PyTorch wheel that fit the tuple (ARM32, Python 3.9), but I've had no luck finding one.

[^1]: From the link given in the article: https://forums.developer.nvidia.com/t/pytorch-for-jetson-version-1-10-now-available/72048
